state_synchronizer/src/coordinator.rs

```diff
@@ -336,9 +336,15 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
             .into_option()
         {
             // node has received a response from peer, so remove peer entry from requests map
-            self.peer_manager.process_response(version, *peer_id);
+            let response = self.peer_manager.process_response(version, *peer_id);
```

Comment:
nit: can we add `has_requested` method to peer_manager instead of having hacky return result. Just for readability